Index: vrpn-wiimote/quat/makefile
===================================================================
--- vrpn-wiimote.orig/quat/makefile
+++ vrpn-wiimote/quat/makefile
@@ -1,9 +1,11 @@
 #############################################################################
-# 
+#
 # makefile for quaternion lib
 #
 #############################################################################
 
+@VRPN_HOST_PLATFORM@
+
 ##########################
 # common definitions. For non-UNC sites, uncomment one of the lines
 # that defines hw_os for the machine you are on in the section just
@@ -218,7 +220,7 @@ qxform : testapps/qxform.c
 	$(CC) -o $(HW_OS)/$@ $(CFLAGS) $(LDFLAGS) testapps/$@.c -lquat -lm
 
 #
-# qmat- matrix to quaternion 
+# qmat- matrix to quaternion
 #
 qmat : qmat.c
 	$(CC) -o $(HW_OS)/$@ $(CFLAGS) $(LDFLAGS) testapps/$@.c -lquat -lm
@@ -241,7 +243,7 @@ oglmult : oglmult.c
 #
 #############################################################################
 
-QUAT_INCLUDES = quat.h 
+QUAT_INCLUDES = quat.h
 QUAT_C_FILES = quat.c matrix.c vector.c xyzquat.c
 QUAT_OBJ_FILES = $(QUAT_C_FILES:.c=.o)
 
@@ -269,4 +271,3 @@ allclean :
 clean :
 	-/bin/rm -f *.o *.a *~ *.j foo a.out $(TEST_FILES)
 	-/bin/rm -fr $(HW_OS)$(OBJECT_DIR_SUFFIX)/*.o $(HW_OS)$(OBJECT_DIR_SUFFIX)/*.a
-
Index: vrpn-wiimote/vrpn/Makefile
===================================================================
--- vrpn-wiimote.orig/vrpn/Makefile
+++ vrpn-wiimote/vrpn/Makefile
@@ -27,6 +27,8 @@
 #     compiled into the server library!
 #############################################################################
 
+@VRPN_HOST_PLATFORM@
+
 ##########################
 # common definitions. For non-UNC sites, uncomment one of the lines
 # that defines hw_os for the machine you are on in the section just
Index: vrpn-wiimote/vrpn/client_src/Makefile
===================================================================
--- vrpn-wiimote.orig/vrpn/client_src/Makefile
+++ vrpn-wiimote/vrpn/client_src/Makefile
@@ -2,6 +2,8 @@
 # Copied the one from ../server_src here and put the former one
 # in Makefile.old
 
+@VRPN_HOST_PLATFORM@
+
 ##########################
 # common definitions. For non-UNC sites, uncomment one of the lines
 # that defines hw_os for the machine you are on in the section just
@@ -335,5 +337,3 @@ clean:
 	rm -f $(OBJ_DIR)/*
 
 $(LIB_DIR)/libvrpn.a:
-
-
Index: vrpn-wiimote/vrpn/server_src/Makefile
===================================================================
--- vrpn-wiimote.orig/vrpn/server_src/Makefile
+++ vrpn-wiimote/vrpn/server_src/Makefile
@@ -1,5 +1,7 @@
 # this is a gmake file
 
+@VRPN_HOST_PLATFORM@
+
 ##########################
 # common definitions. For non-UNC sites, uncomment one of the lines
 # that defines hw_os for the machine you are on in the section just
@@ -87,11 +89,11 @@ endif
 
 ifeq ($(HW_OS),pc_linux_ia64)
 	CC := g++
-	SYSLIBS := 
+	SYSLIBS :=
 endif
 ifeq ($(HW_OS),pc_linux64)
 	CC := g++ -m64
-	SYSLIBS := 
+	SYSLIBS :=
 endif
 ifeq ($(HW_OS), pc_linux_arm)
 	CC := arm-linux-g++
@@ -104,7 +106,7 @@ ifeq ($(HW_OS), pc_cygwin_arm)
 	AR := arm-unknown-linux-gnu-ar ruv
 endif
 ifeq ($(HW_OS),pc_linux)
-	SYSLIBS := 
+	SYSLIBS :=
 # For WiiMote
 #	SYSLIBS := -lwiiuse
 # For GPM mouse
@@ -113,7 +115,7 @@ ifeq ($(HW_OS),pc_linux)
 #	SYSLIBS := -ldl
 endif
 ifeq ($(HW_OS),pc_linux64)
-	SYSLIBS := 
+	SYSLIBS :=
 # For WiiMote
 #	SYSLIBS := -lwiiuse
 # For GPM mouse
@@ -272,7 +274,7 @@ $(OBJ_DIR)/vrpn_server: $(OBJ_DIR)/vrpn.
 		$(OBJ_DIR)/texture_plane.o $(OBJ_DIR)/trimesh.o \
 		$(OBJ_DIR)/vrpn_Phantom.o \
 		$(OBJ_DIR)/InstantBuzzEffect.o \
-		$(OBJ_DIR)/vrpn_Generic_server_object.o 
+		$(OBJ_DIR)/vrpn_Generic_server_object.o
 	$(CC) $(LFLAGS) -o $(OBJ_DIR)/vrpn_server $(OBJ_DIR)/vrpn.o \
 		$(OBJ_DIR)/buzzForceField.o $(OBJ_DIR)/constraint.o \
 		$(OBJ_DIR)/forcefield.o $(OBJ_DIR)/plane.o \
@@ -393,4 +395,3 @@ $(OBJ_DIR)/sample_server.o:	sample_serve
 $(OBJ_DIR)/test_mutex.o:	test_mutex.C
 $(OBJ_DIR)/test_peerMutex.o:	test_peerMutex.C
 $(OBJ_DIR)/sample_analog.o:	sample_analog.C
-
