Index: vrpn-07.22/vrpn/Makefile
===================================================================
--- vrpn-07.22.orig/vrpn/Makefile
+++ vrpn-07.22/vrpn/Makefile
@@ -6,7 +6,7 @@
 # On the sgi, both g++ and CC verisons are built by default.
 #
 # Author: Russ Taylor, 10/2/1997
-#	  
+#
 # modified:
 # * Jeff Juliano, 10/99
 #    * added "make depend"  (see comments at end of this makefile)
@@ -140,7 +140,7 @@ else
         CC := gcc
         RANLIB := ranlib
   endif
-  
+
   ifeq ($(HW_OS), powerpc_macosx)
         CC := cc
         RANLIB := ranlib
@@ -187,7 +187,7 @@ else
 	CC := /usr/local/lib/CenterLine/bin/CC
   endif
   ifeq ($(HW_OS), hp_flow_aCC)
-	CC := /opt/aCC/bin/aCC 
+	CC := /opt/aCC/bin/aCC
   endif
 endif
 
@@ -230,12 +230,12 @@ SYS_INCLUDE :=
 
 ifeq ($(HW_OS),powerpc_macosx)
 #  SYS_INCLUDE := -I/usr/include
-   SYS_INCLUDE :=-DMACOSX -I../isense 
+   SYS_INCLUDE :=-DMACOSX -I../isense
 endif
 
 ifeq ($(HW_OS),universal_macosx)
 #  SYS_INCLUDE := -I/usr/include
-   SYS_INCLUDE :=-DMACOSX -I../isense 
+   SYS_INCLUDE :=-DMACOSX -I../isense
 endif
 
 ifeq ($(HW_OS),pc_linux)
@@ -262,17 +262,17 @@ ifeq ($(HW_OS),sgi_irix)
   SYS_INCLUDE :=
 endif
 
-ifeq ($(HW_OS),hp700_hpux10) 
+ifeq ($(HW_OS),hp700_hpux10)
   SYS_INCLUDE := -I/usr/local/contrib/include -I/usr/local/contrib/mod/include \
                  -I/usr/include/bsd
 endif
 
-ifeq ($(HW_OS),hp_flow) 
+ifeq ($(HW_OS),hp_flow)
   SYS_INCLUDE := -I/usr/local/contrib/include -I/usr/local/contrib/mod/include \
                  -I/usr/include/bsd -DFLOW
 endif
 
-ifeq ($(HW_OS),hp_flow_aCC) 
+ifeq ($(HW_OS),hp_flow_aCC)
   SYS_INCLUDE := -I/usr/local/contrib/include -I/usr/local/contrib/mod/include \
                  -I/usr/include/bsd -DFLOW
 endif
@@ -394,6 +394,8 @@ $(AOBJECT_DIR)/%.o: %.C $(ALIB_INCLUDES)
 # If we're building for sgi_irix, we need both g++ and non-g++ versions,
 # unless we're building for one of the weird ABIs, which are only supported
 # by the native compiler.
+all:
+	$(MAKE) -C ../quat
 
 ifeq ($(HW_OS),sgi_irix)
   ifeq ($(SGI_ABI),32)
@@ -419,6 +421,13 @@ server_g++:
 	$(MAKE) FORCE_GPP=1 $(UNQUAL_OBJECT_DIR)/g++/libvrpnserver.a
 	$(MV) $(UNQUAL_OBJECT_DIR)/g++/libvrpnserver.a $(UNQUAL_OBJECT_DIR)/libvrpnserver_g++.a
 
+SUBDIRS = client_src server_src
+.PHONY: subdirs $(SUBDIRS)
+all: subdirs
+subdirs: $(SUBDIRS)
+$(SUBDIRS):
+	$(MAKE) -C $@
+
 .PHONY:	client
 client: $(OBJECT_DIR) $(OBJECT_DIR)/libvrpn.a
 
@@ -468,7 +477,7 @@ LIB_INCLUDES = vrpn_Connection.h vrpn_Tr
 		vrpn_BaseClass.h vrpn_Imager.h \
 		vrpn_Analog_Output.h vrpn_Poser.h vrpn_Auxiliary_Logger.h
 
-$(LIB_OBJECTS): 
+$(LIB_OBJECTS):
 $(OBJECT_DIR)/libvrpn.a: $(MAKEFILE) $(LIB_OBJECTS)
 	$(AR) $(OBJECT_DIR)/libvrpn.a $(LIB_OBJECTS)
 	-$(RANLIB) $(OBJECT_DIR)/libvrpn.a
@@ -523,7 +532,7 @@ SLIB_INCLUDES = $(LIB_INCLUDES) vrpn_3Sp
 	vrpn_Tracker_MotionNode.h vrpn_Tracker_NDI_Polaris.h \
 	vrpn_WiiMote.h vrpn_Tracker_Isotrak.h
 
-$(SLIB_OBJECTS): 
+$(SLIB_OBJECTS):
 $(OBJECT_DIR)/libvrpnserver.a: $(MAKEFILE) $(SLIB_OBJECTS)
 	$(AR) $(OBJECT_DIR)/libvrpnserver.a $(SLIB_OBJECTS)
 	-$(RANLIB) $(OBJECT_DIR)/libvrpnserver.a
@@ -539,7 +548,7 @@ ALIB_OBJECTS = $(patsubst %,$(AOBJECT_DI
 ALIB_INCLUDES =  \
 	vrpn_atmellib.h vrpn_atmellib_helper.h vrpn_atmellib_errno.h
 
-$(ALIB_OBJECTS): 
+$(ALIB_OBJECTS):
 $(OBJECT_DIR)/libvrpnatmel.a: $(MAKEFILE) $(ALIB_OBJECTS)
 	$(AR) $(OBJECT_DIR)/libvrpnatmel.a $(ALIB_OBJECTS)
 	-$(RANLIB) $(OBJECT_DIR)/libvrpnatmel.a
@@ -598,6 +607,11 @@ install: all
 	( cd $(OBJECT_DIR) ; cp *.a $(LIB_DIR) )
 	-mkdir $(INCLUDE_DIR)
 	cp vrpn*.h $(INCLUDE_DIR)
+	for dir in $(SUBDIRS); do \
+		$(MAKE) -C $$dir install; \
+	done
+	cp ../quat/quat.h $(INCLUDE_DIR)
+
 
 uninstall:
 	( cd $(LIB_DIR) ; rm -f libvrpn*.a )
